### ISSUE TITLE
restructure: Change qSM_Signal_t from qUINT32_t to struct with Type and Data attributes

### ISSUE THREAD

**Issue by MichalRostekRostkowski** [kmilo17pet/QuarkTS#46]
The suggestion here is to replace current qSM_Signal_t implementation as qUINT32_t with structure with two attributes:
```C
typedef struct {
  qUINT32_t Type;
  qUINT32_t *Data;
} qSM_Signal_t;
```
The idea behind this is to allow HSMs to send signals with data. Certainly it's not just a modification but it would rather require restructure of the whole FSM module. Anyway it is worth discussion.

**Comment by kmilo17pet** [comment]
```
typedef struct {
  qSM_SigID_t  id; 
  void *data;
} qSM_Signal_t;
```
I like the idea but it involves a refactor of the entire FSM module and the way transitions are specified.
Will be on the roadmap, I need to review in detail all the implications.
@thejbte @ManuelPosada opinions?

**Comment by MichalRostekRostkowski** [comment]
Hello guys,
I have been working on some draft for this feature.
Take a look: https://github.com/MichalRostekRostkowski/QuarkTS/pull/4

**Comment by kmilo17pet** [comment]
feature included on v7.3.1
https://github.com/kmilo17pet/QuarkTS/releases/tag/v7.3.1
